Yubico.YubiKey/src/Yubico/YubiKey/Piv/PivAlgorithmExtensions.cs

```diff
@@ -12,13 +12,168 @@
     public static class PivAlgorithmExtensions
     {
+        // Mapping between PIV algorithms and key definitions
+        private static readonly Dictionary<PivAlgorithm, PivAlgorithmDefinition> _pivAlgorithmMap;
+        private static readonly Dictionary<KeyType, PivAlgorithm> _keyTypeToPivAlgorithmMap;
+
+        static PivAlgorithmExtensions()
+        {
+            // Initialize mappings
+            _pivAlgorithmMap = new Dictionary<PivAlgorithm, PivAlgorithmDefinition>
+            {
+                { PivAlgorithm.Rsa1024, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Rsa1024, KeyDefinition = KeyDefinitions.RSA1024 } },
+                { PivAlgorithm.Rsa2048, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Rsa2048, KeyDefinition = KeyDefinitions.RSA2048 } },
+                { PivAlgorithm.Rsa3072, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Rsa3072, KeyDefinition = KeyDefinitions.RSA3072 } },
+                { PivAlgorithm.Rsa4096, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Rsa4096, KeyDefinition = KeyDefinitions.RSA4096 } },
+                { PivAlgorithm.EccP256, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.EccP256, KeyDefinition = KeyDefinitions.P256 } },
+                { PivAlgorithm.EccP384, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.EccP384, KeyDefinition = KeyDefinitions.P384 } },
+                { PivAlgorithm.EccP521, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.EccP521, KeyDefinition = KeyDefinitions.P521 } },
+                { PivAlgorithm.EccEd25519, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.EccEd25519, KeyDefinition = KeyDefinitions.Ed25519 } },
+                { PivAlgorithm.EccX25519, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.EccX25519, KeyDefinition = KeyDefinitions.X25519 } },
+                { PivAlgorithm.Aes128, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Aes128, KeyDefinition = KeyDefinitions.AES128 } },
+                { PivAlgorithm.Aes192, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Aes192, KeyDefinition = KeyDefinitions.AES192 } },
+                { PivAlgorithm.Aes256, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Aes256, KeyDefinition = KeyDefinitions.AES256 } },
+                { PivAlgorithm.TripleDes, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.TripleDes, KeyDefinition = KeyDefinitions.TripleDes } },
+                { PivAlgorithm.None, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.None, KeyDefinition = null! } },
+                { PivAlgorithm.Pin, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Pin, KeyDefinition = null! } },
```

Comment:
Could PivAlgorithm.None and PivAlgorithm.Pin be removed from the map?

Reply:
I wonder what happens when you call one of the extension functions on .Pin or .None?

```
// public static bool IsRsa(this PivAlgorithm algorithm)
var isRsa = PivAlgorithm.Pin.IsRsa();
```

The extension function directly accesses the `KeyDefinition.IsRsaKey` property and for both `.Pin` and `.None` the KeyDefinition is `null!` so I would expect a runtime exception is thrown, but I would expect to get a false result.

Maybe it would be better of assigning `null!` to have a special KeyDefinition for .None and .Pin which would return the correct values.